File: Insertion_Sort/insertion_sort.cpp
#include <iostream>

void insertion_sort(int l[], int n)
{
    for(int i = 0; i < n - 1; i++)
    {
        int j = i;
        while (l[j] > l[j + 1] && j >= 0)
        {
            std::swap(l[j], l[j + 1]);
            j--;
        }
    }
}


// A utility function to print an array of size n
void print_array(int l[], int n)
{
    int i;
    for (i = 0; i < n; i++)
        std::cout << l[i] << " ";
    std::cout << std::endl;
}

// Driver code to test above
int main()
{
    int arr[] = { 12, 11, 13, 5, 6 };
    int n = sizeof(arr) / sizeof(arr[0]);

    insertion_sort(arr, n);
    print_array(arr, n);

    return 0;
}

File: Insertion_Sort/InsertionSort.java
public class InsertionSort {
    
    // A utility function to print an array
    public static void printArray(int[] arr) {
        for(int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }
    
    public static void insertionSort(int[] arr) {
        for(int i = 0; i < arr.length - 1; i++) {
            int j = i;
            while(j >= 0 && arr[j] > arr[j+1]) {
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;  
                j--;
            }
        }
    }
    
    // Driver code to test above
    static int[] arr = {6, 1, 5, 9, 9, 5, 3, 8, 0};
    
    public static void main(String[] args) {
        insertionSort(arr);
        printArray(arr);
    }
        
}